r/R/dataset-format.R

```diff
@@ -309,14 +309,52 @@ FileWriteOptions <- R6Class("FileWriteOptions",
   inherit = ArrowObject,
   public = list(
     update = function(table, ...) {
+      compose_err_header <- function(unsupported_passed_args) {
+        paste0(
+          oxford_paste(unsupported_passed_args),
+          ngettext(length(unsupported_passed_args),
+                   " is not a valid argument ",
+                   " are not valid arguments "),
+          "for your chosen `format`."
```

Comment:
could we? should we? also offer what the supported arguments are (if we have them as a list already...)

Reply:
We could, I don't think we should. Or, at least, we shouldn't until we can do this in a robust way (i.e. for `type == "ipc"`  they are currently hard-coded)

Reply:
Also, the user might not be aware that when `type %in% c("parquet", "arrow", "ipc")` implementation happens through `IpcFileWriteOptions` and once we are inside `FileWriteOptions$update()` we are unaware of the exact value passed for `format` (as it has been transformed into a `FileFormat` object). Hence my somewhat ambiguous formulation `"your chose format"`. We could try to print supported `args`, but that might make it a bit more confusing and difficult to keep in sync with the function docs.